src/odemis/gui/comp/grid.py

```diff
@@ -97,9 +97,22 @@ def hide_all_viewports(self):
         """ Hide all viewports """
         self.set_shown_viewports()
 
+    def get_4_viewports(self):
```

Comment:
This function has a big side-effect: it will force the viewports to be Shown. It's really not a good idea if the function is called `get...()`. You have two possibilities: change the function name to be more obvious (ex, "show_4_viewports"), or remove this side effect.
I'd suggest the second alternative, and only in the places you really want to ensure the 4 viewports are shown, after calling `get_4_viewports()`, you call `v.Show()` for each of them.

Reply:
Since the grid handles the visibility of the viewports, I removed the "show" functionality from this function so that it can be entirely handled by the existing functions. Seemed safer.